### ISSUE TITLE
Implement chunking to transfer bulk data

### ISSUE THREAD

**Issue by miticollo** [miticollo/frida-ios-dump#5]
This issue  is for Linux users because it seems that macOS users have no problems.

For more detail see [here](https://github.com/frida/frida-gum/issues/370#issuecomment-549190480) and [here](https://t.me/fridadotre/29673).

**Comment by Epic-Oreo** [comment]
Im having this issue on mac, I dont know how to fix it (if i can)

**Comment by miticollo** [comment]
- Did you use the chunking branch?
- What app?

**Comment by Epic-Oreo** [comment]
I was not using the chunking branch but now i am. ~~Am getting difrent error of 
`Compilation failed: [{'category': 'error', 'code': 2307, 'file': {'path': '/Users/{name}/Documents/Ipa/frida-ios-dump/agent/lib/pull.ts', 'line': 0, 'character': 47}, 'text': "Cannot find module 'frida-remote-stream' or its corresponding type declarations."}]
Failed to decrypt: compilation failed`~~

Figured that out as you wrote that comment :) thank you :)

im trying to do youtube

**Comment by miticollo** [comment]
Open a new terminal in `agent` folder and run `npm -ddd install` to install locally every NodeJS dependencies.
Run script again.

**Comment by LaiKash** [comment]
Hi! I think I am facing the same exact issue. It's a huge APP so it might be expected, with smaller ones it works:

```frida-ios-dump git:(master) python ./decrypter.py -f com.ubercab.UberPartner -U              
Injecting script...                                                     

(process:48681): GLib-GIO-WARNING **: 14:42:48.447: _g_dbus_worker_do_read_cb: error determining bytes needed: Blob indicates that message exceeds maximum message length (128MiB)
the connection is closed
```

Edit: I was not using the chunking branch but it does't work with that one either :)

**Comment by miticollo** [comment]
> Hi! I think I am facing the same exact issue. It's a huge APP so it might be expected, with smaller ones it works:
> 
> ```frida-ios-dump git:(master) python ./decrypter.py -f com.ubercab.UberPartner -U              
> Injecting script...                                                     
> 
> (process:48681): GLib-GIO-WARNING **: 14:42:48.447: _g_dbus_worker_do_read_cb: error determining bytes needed: Blob indicates that message exceeds maximum message length (128MiB)
> the connection is closed```

- Did you use the chunking branch?
- What app?

**Comment by LaiKash** [comment]
> > Hi! I think I am facing the same exact issue. It's a huge APP so it might be expected, with smaller ones it works:
> > ```
> > Injecting script...                                                     
> > 
> > (process:48681): GLib-GIO-WARNING **: 14:42:48.447: _g_dbus_worker_do_read_cb: error determining bytes needed: Blob indicates that message exceeds maximum message length (128MiB)
> > the connection is closed```
> > ```
> 
> * Did you use the chunking branch?
> * What app?

Yes! Using the chunking branch. The APP is Uber Driver (it's huge, I know)

**Comment by miticollo** [comment]
I decrypted with macOS 13.6.7 three different versions of this app installed on an iPhone 8+ (iOS 16.7.5), iPhone 7 (iOS 13.6.1) and iPhone SE 2020 (iOS 14.4.2). And frida-ios-dump always transfered all files without any problem.
